internal/common/common.go

```diff
@@ -113,17 +114,68 @@ func GetRootCredentials(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
+func Getgitmeta(clusterName string) (gitopsRepoName string, metaphorRepoName string) {
+	var gitopsFound, metaphorFound bool
+	homePath,err := os.UserHomeDir()
+	dirs, err := ioutil.ReadDir(fmt.Sprintf("%s/.k1/%s", homePath, clusterName))
+	if err != nil {
+		log.Info().Msg("Error reading directory")
+		return "cantfindgit","cantfindmeta"
+	}
+
+	for _, direc := range dirs {
+		if direc.IsDir() {
+			parentdir, err := ioutil.ReadDir(fmt.Sprintf("%s/.k1/%s/%s", homePath, clusterName, direc.Name()))
+			if err != nil {
+				log.Printf("Error reading directory %s: %v", direc.Name(), err)
+				continue
+			}
+
+			for _, dir := range parentdir {
+				if dir.IsDir() {
+					if dir.Name() == "registry" {
+						gitopsRepoName = direc.Name()
+						gitopsFound = true
+					} else
+					if dir.Name() == ".github" {
+						metaphorRepoName = direc.Name()
+						metaphorFound = true
+					}
+				}
+			}
+
+
+		}
+	}
+	
+	
+
+	if !gitopsFound {
+		log.Info().Msg("Gitops Repo not found")
+		return "cantfindgit","cantfindmeta"
+	}
+
+	if !metaphorFound {
+		log.Info().Msg("Metaphor Repo not found")
+		os.Exit(1)
```

Comment:
Happy to chat about this one, but this seems wrong: exiting right here means that several pieces of the Go code here won't be executed (like `defer` or similar).

It's better to return errors that the callers can decide whether or not they need to make an action out of it. As a golden rule, the only place that should exit with an exit code is the top-most `main()` function from the Go application. As a package (called `common` here) it's better to return the error so the caller handles it and decides if it's so critical that the app needs to exit and crash.